docker/resource_docker_plugin.go

```diff
@@ -0,0 +1,67 @@
+package docker
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceDockerPlugin() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDockerPluginCreate,
+		Read:   resourceDockerPluginRead,
+		Update: resourceDockerPluginUpdate,
+		Delete: resourceDockerPluginDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"plugin_reference": {
```

Comment:
keep consistent with the other resources and have this attribute as `name` rather than `plugin_reference`

Reply:
I concern `name` is misunderstood as `alias`.

ex.

```
❯ docker plugin ls
ID             NAME                          DESCRIPTION                         ENABLED
2c6c299f4899   sample-volume-plugin:latest   A sample volume plugin for Docker   false
```

Reply:
would 'image' be better?

Reply:
Might be so.
@mavogel Do you have any opinion?

Reply:
I suggest using `name` to be also consistent with the json output of `docker plugin inspect`

Reply:
@suzuki-shunsuke I think this is also still open

Reply:
@mavogel 

> I suggest using name to be also consistent with the json output of docker plugin inspect

`name` of `docker plugin inspect` is different from `plugin_reference`.

```
$ docker plugin install --alias foo tiborvass/sample-volume-plugin
$ docker plugin inspect foo | jq -r ".[0].PluginReference"                                                                                                                                                      
docker.io/tiborvass/sample-volume-plugin:latest
$ docker plugin inspect foo | jq -r ".[0].Name"                                                                                                                                                                 
foo:latest
```

```
$ docker version                                                                                                                                                                                                
Client: Docker Engine - Community                                                                                                                                                                               
 Cloud integration: 1.0.4                                                                                                                                                                                       
 Version:           20.10.2                                                                                                                                                                                     
 API version:       1.41                                                                                                                                                                                        
 Go version:        go1.13.15                                                                                                                                                                                   
 Git commit:        2291f61                                                                                                                                                                                     
 Built:             Mon Dec 28 16:12:42 2020                                                                                                                                                                    
 OS/Arch:           darwin/amd64                                                                                                                                                                                
 Context:           default                                                                                                                                                                                     
 Experimental:      true                                                                                                                                                                                        
                                                                                                                                                                                                                
Server: Docker Engine - Community                                                                                                                                                                               
 Engine:                                                                                                                                                                                                        
  Version:          20.10.2                                                                                                                                                                                     
  API version:      1.41 (minimum version 1.12)                                                                                                                                                                 
  Go version:       go1.13.15                                                                                                                                                                                   
  Git commit:       8891c58                                                                                                                                                                                     
  Built:            Mon Dec 28 16:15:23 2020                                                                                                                                                                    
  OS/Arch:          linux/amd64                                                                                                                                                                                 
  Experimental:     true                                                                                                                                                                                        
 containerd:                                                                                                                                                                                                    
  Version:          v1.4.3                                                                                                                                                                                      
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b                                                                                                                                                    
 runc:                                                                                                                                                                                                          
  Version:          1.0.0-rc92                                                                                                                                                                                  
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff                                                                                                                                                    
 docker-init:                                                                                                                                                                                                   
  Version:          0.19.0                                                                                                                                                                                      
  GitCommit:        de40ad0
```

```json
[
    {
        "Config": {
            "Args": {
                "Description": "",
                "Name": "",
                "Settable": null,
                "Value": null
            },
            "Description": "A sample volume plugin for Docker",
            "Documentation": "https://docs.docker.com/engine/extend/plugins/",
            "Entrypoint": [
                "/usr/bin/sample-volume-plugin",
                "/data"
            ],
            "Env": [
                {
                    "Description": "",
                    "Name": "DEBUG",
                    "Settable": [
                        "value"
                    ],
                    "Value": "0"
                }
            ],
            "Interface": {
                "Socket": "plugin.sock",
                "Types": [
                    "docker.volumedriver/1.0"
                ]
            },
            "IpcHost": false,
            "Linux": {
                "AllowAllDevices": false,
                "Capabilities": null,
                "Devices": null
            },
            "Mounts": null,
            "Network": {
                "Type": ""
            },
            "PidHost": false,
            "PropagatedMount": "/data",
            "User": {},
            "WorkDir": "",
            "rootfs": {
                "diff_ids": [
                    "sha256:be0f976063f493d2d3a36e831e62aecbae9ecbaf142eca24a274914b79b96d4a"
                ],
                "type": "layers"
            }
        },
        "Enabled": true,
        "Id": "15868b05e7dc5cd20eb9264b0c781f1f9ba8258e65f3976f703977e05f8814cf",
        "Name": "foo:latest",
        "PluginReference": "docker.io/tiborvass/sample-volume-plugin:latest",
        "Settings": {
            "Args": [],
            "Devices": [],
            "Env": [
                "DEBUG=0"
            ],
            "Mounts": []
        }
    }
]
```

Reply:
I don't think I would worry about the idiosyncrasies of the API. The other resources use name to refer to the image to fetch and as a UX point it would be good to be consistent.

Reply:
I added some changes.

7a7aee3025ac7893b3b81638559cc669da077e29
0950bba77c34478329f2c5f34c3a89913c1e3356

* add `DiffSuppressFunc` to alias to complement the latest tag
* add a required field `name`
  * normalize `name` with [reference.ParseAnyReference](https://pkg.go.dev/github.com/docker/distribution/reference#ParseAnyReference) and complement the latest tag
* make `plugin_reference` read only